module/zfs/vdev.c

```diff
@@ -1848,6 +1878,25 @@ vdev_open(vdev_t *vd)
 		 */
 		vd->vdev_asize = asize;
 		vd->vdev_max_asize = max_asize;
+
+		/*
+		 * If the vdev_ashift was not overriden at creation time,
+		 * then set it the logical ashift and optimize the ashift.
+		 */
+		if (vd->vdev_ashift == 0) {
+			vd->vdev_ashift = MAX(
+			    vd->vdev_logical_ashift, vd->vdev_ashift);
+
```

Comment:
just reading the code, that MAX() seem useless

Reply:
I will remove. Thanks!